ci/prow/make_config.go

```diff
@@ -996,9 +1190,308 @@ func (a *stringArrayFlag) Set(value string) error {
+			}
+
+			// TODO: confirm if they are needed or not
+			if jobName == "latency" {
+				extras["short_text_metric"] = "latency"
+			}
+			if jobName == "api-coverage" {
+				extras["short_text_metric"] = "api_coverage"
+			}
+			if jobName == "performance" {
+				extras["short_text_metric"] = "perf_latency"
+			}
+		case "nightly":
+			gcsLogDir = fmt.Sprintf("%s/%s/ci-%s-%s-release", gcsBucket, logsDir, repoName, jobName)
+		case "test-coverage":
+			gcsLogDir = fmt.Sprintf("%s/%s/ci-%s-%s", gcsBucket, logsDir, repoName, "go-coverage")
+			extras["short_text_metric"] = "coverage"
+		default:
+			continue
+		}
+		executeTestGroupTemplate(testGroupName, gcsLogDir, extras)
+	}
+}
+
+// executeTestGroupTemplate outputs the given test group config template with the given data
+func executeTestGroupTemplate(testGroupName string, gcsLogDir string, extras map[string]string) {
+	var data testGroupTemplateData
+	data.Base.TestGroupName = testGroupName
+	data.GcsLogDir = gcsLogDir
+	data.Extras = extras
+	executeTemplate("test group", testGroupTemplate, data)
+}
+
+// generateDashboard generates the dashboard configuration
+func generateDashboard(repoName string, jobNames []string) {
+	outputConfig(fmt.Sprintf("- name: %s\n  dashboard_tab:", repoName))
+	for _, jobName := range jobNames {
+		testGroupName := getTestGroupName(repoName, jobName)
+		baseOptions := "sort-by-name="
+		extras := make(map[string]string)
+		switch jobName {
+		case "continuous":
+			dashboardTabName := jobName
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+
+			if repoName == "knative-serving" {
+				dashboardTabName := "conformance-tests"
+				baseOptions = "include-filter-by-regex=test/conformance\\\\.&sort-by-name="
+				executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+			}
+		case "dot-release", "auto-release", "performance", "latency", "api-coverage", "playground":
+			dashboardTabName := jobName
+
+			if jobName == "latency" || jobName == "api-coverage" {
+				baseOptions = "exclude-filter-by-regex=Overall$&group-by-directory=&expand-groups=&sort-by-name="
+			}
+			if jobName == "performance" {
+				baseOptions = "exclude-filter-by-regex=Overall$&group-by-target=&expand-groups=&sort-by-name="
+			}
+			if jobName == "latency" {
+				extras["description"] = "95% latency in ms"
+			}
+			if jobName == "api-coverage" {
+				extras["description"] = "Conformance tests API coverage."
+			}
+
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+		case "nightly":
+			dashboardTabName := "release"
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+		case "test-coverage":
+			dashboardTabName := "coverage"
+			baseOptions = "exclude-filter-by-regex=Overall$&group-by-directory=&expand-groups=&sort-by-name="
+			executeDashboardTabTemplate(dashboardTabName, testGroupName, baseOptions, extras)
+		default:
+			continue
```

Comment:
Shouldn't you fail here due to an unknown job name?

Reply:
Added fatal log.